slash/core/fixtures/fixture_store.py

```diff
@@ -255,25 +255,35 @@ def get_fixture_by_id(self, fixture_id):
 
     def get_fixture_dict(self, fixture_names, namespace=None, get_values=True, skip_names=frozenset()):
         returned = {}
-
         if namespace is None:
             namespace = self.get_current_namespace()
+        fixture_list = self._get_fixture_list(fixture_names, skip_names=skip_names, namespace=namespace)
+        for required_name, fixture in fixture_list:
+            if get_values:
+                fixture = self.get_fixture_value(fixture, name=required_name)
+            if required_name in fixture_names or [element for element in fixture_names if required_name in element]:
+                returned[required_name] = fixture
+        return returned
 
+    def _get_fixture_list(self, fixture_names, namespace=None, skip_names=frozenset(), fixtures_list=None):
+        if fixtures_list is None:
+            fixtures_list = []
         for element in fixture_names:
             if isinstance(element, tuple):
                 required_name, real_name = element
             else:
                 required_name = real_name = element
-
             if required_name in skip_names:
                 continue
             if element == 'this':
                 continue
             fixture = namespace.get_fixture_by_name(real_name)
-            if get_values:
-                fixture = self.get_fixture_value(fixture, name=required_name)
-            returned[required_name] = fixture
-        return returned
+            if (required_name, fixture) not in fixtures_list:
```

Comment:
I think fixture_list should actually be a set. I'm not 100% sure if this has implications, but this nested lookup can be slow in extreme cases...